mantidimaging/gui/windows/live_viewer/test/model_test.py

```diff
@@ -0,0 +1,140 @@
+            self.mock_signal_image = mock.create_autospec(pyqtSignal, emit=mock.Mock())
+            self.watcher.image_changed = self.mock_signal_image
+
+    def _make_simple_dir(self, directory: Path, t0: float = 1000):
+        file_list = [directory / f"abc_{i:06d}.tif" for i in range(5)]
+        if not directory.exists():
+            self.fs.create_dir(directory)
+        os.utime(directory, (10, t0))
+        n = 1
+        for file in file_list:
+            self.fs.create_file(file)
+            os.utime(file, (10, t0 + n))
+            n += 1
+
+        return file_list
+
+    def _make_sub_directories(self, directory: Path, sub_dirs: list[str], t0: float = 1000):
+        n = 0
+        for sub_dir in sub_dirs:
+            self.fs.create_dir(directory / sub_dir)
+            os.utime(directory / sub_dir, (10, t0 + n))
+            n += 1
+            file_list = [directory / sub_dir / f"abc_{i:06d}.tiff" for i in range(5)]
+            for file in file_list:
+                self.fs.create_file(file)
+                os.utime(file, (10, t0 + n))
+                n += 1
+
+        return file_list
+
+    def _get_recent_emitted_files(self):
+        self.mock_signal_image.emit.assert_called()
+        last_call_args = self.mock_signal_image.emit.call_args_list[-1]
+        emitted_images = [image_data.image_path for image_data in last_call_args[0][0]]
+        return emitted_images
+
+    def test_WHEN_find_images_called_THEN_returns_images(self):
+        file_list = self._make_simple_dir(self.top_path)
+
+        images_datas = self.watcher.find_images(self.top_path)
+
+        images = [image.image_path for image in images_datas]
+        self._file_list_count_equal(images, file_list)
+
+    def test_WHEN_find_images_deleted_file_THEN_handles_error(self):
+        # Simulate the case where a file returned by iterdir has been deleted when we come to read it
+        file_list = self._make_simple_dir(self.top_path)
+        self.fs.remove(file_list[0])
+
+        # mocked iterdir() gives full list, but first file as been deleted
+        mock_directory = mock.create_autospec(Path, iterdir=lambda: file_list)
+
+        images_datas = self.watcher.find_images(mock_directory)
+
+        images = [image.image_path for image in images_datas]
+        self._file_list_count_equal(images, file_list[1:])
+
+    def test_WHEN_find_sub_directories_called_THEN_finds_subdirs(self):
+        self._file_in_sequence(self.top_path, self.watcher.sub_directories.keys())
+        self.assertEqual(len(self.watcher.sub_directories), 1)
+        subdir_names = ["tomo", "flat_before", "flat_after"]
+        self._make_sub_directories(self.top_path, subdir_names)
+
+        self.watcher.find_sub_directories(self.top_path)
+        for sub_dir in subdir_names:
+            self._file_in_sequence(self.top_path / sub_dir, self.watcher.sub_directories.keys())
+
+    def test_WHEN_directory_change_simple_THEN_images_emitted(self):
```

Comment:
Do you agree that it may be worth us adding an assertion prior to adding the simple_dir to make sure no file change signal has been emitted when we don't expect one?